tests/cases/fourslash/server/renameNamedImport.ts

```diff
@@ -0,0 +1,22 @@
+/// <reference path='../fourslash.ts' />
+
+// @Filename: /lib/tsconfig.json
+//// {}
+
+// @Filename: /lib/index.ts
+//// const unrelatedLocalVariable = 123;
+//// export const someExportedVariable = unrelatedLocalVariable;
+
+// @Filename: /src/tsconfig.json
+//// {}
+
+// @Filename: /src/index.ts
+//// import { /*i*/someExportedVariable } from '../lib/index';
```

Comment:
the bug (incorrectly renaming the original thing and not just the local import) also happened for named imports, besides namespace imports.

Reply:
So, I don’t think this one should have changed, or at least, the change here is incomplete. The big difference between import specifiers and every other kind of import alias is that import specifiers must reflect the name of something being exported. If you rename this without renaming the export, you’ll get an error. A default or namespace import is freely nameable. The only ways you can correctly rename an import specifier are by renaming the corresponding export or introducing a local name binding, e.g. `someExportedVariable as foo`. Which one you want to do is subjective/circumstantial, so I’m not sure what the right solution is. We have a preference for the same issue with object literal shorthand properties:

<img width="907" alt="image" src="https://user-images.githubusercontent.com/3277153/164112539-c5f68bdc-5879-4c30-afa7-40cc7df5bf2c.png">

I believe this controls whether a rename on

```ts
const x = 0;
const obj = { x/*HERE*/ };
```

becomes

```ts
const renamed = 0;
const obj = { renamed };
```

or

```ts
const x = 0;
const obj = { renamed: x };
```

which are essentially the two options I just outlined for import specifiers, but cross-file. I have never liked this preference because it’s not a _user_ preference per se; sometimes I want one and sometimes I want the other, and it’s annoying that I can’t select the behavior without editing my preferences. Unless we’re willing to take the time to invest in a better solution, I would suggest not changing the import specifier behavior.